app/frontend/courses/curricula/components/CoursesCurriculum__ModerationReportButton.res

```diff
@@ -0,0 +1,135 @@
+open CoursesCurriculum__Types
+
+module CreateModerationReportMutation = %graphql(`
+   mutation CreateModerationReportMutation($reason: String!, $reportableId: String!, $reportableType: String! ) {
+     createModerationReport(reason: $reason, reportableId: $reportableId, reportableType: $reportableType ) {
+       moderationReport {
+         id
+         reason
+         reportableId
+         reportableType
+         userId
+       }
+     }
+   }
+   `)
+
+let createModerationReport = (
+  reportableType,
+  reportableId,
+  reason,
+  setModerationReports,
+  setShowReport,
+  setReportReason,
+  event,
+) => {
+  ReactEvent.Mouse.preventDefault(event)
+  CreateModerationReportMutation.make({
+    reason,
+    reportableId,
+    reportableType,
+  })
+  |> Js.Promise.then_(response => {
+    switch response["createModerationReport"]["moderationReport"] {
+    | Some(moderationReport) =>
+      setModerationReports(moderationReports =>
+        Js.Array2.concat([moderationReport], moderationReports)
+      )
+      setShowReport(_ => false)
+      setReportReason(_ => "")
+    | None => ()
+    }
+    Js.Promise.resolve()
+  })
+  |> ignore
+}
+
+let updateReportReason = (setReportReason, event) => {
+  ReactEvent.Form.preventDefault(event)
+  setReportReason(ReactEvent.Form.currentTarget(event)["value"])
+}
+
+let updateShowReport = (setShowReport, showReport, event) => {
+  ReactEvent.Mouse.preventDefault(event)
+  setShowReport(_ => showReport)
+}
+
+@react.component
+let make = (~currentUser, ~moderationReports, ~reportableId, ~reportableType) => {
+  let (moderationReports, setModerationReports) = React.useState(() => moderationReports)
+  let (showReport, setShowReport) = React.useState(() => false)
+  let (reportReason, setReportReason) = React.useState(() => "")
+
+  let reported = Belt.Array.reduce(moderationReports, false, (acc, report) =>
+    acc || report->ModerationReport.userId === currentUser->CurrentUser.id
+  )
+
+  <div className="relative">
+    {showReport
+      ? <dialog
+          role="dialog"
+          className="fixed inset-0 z-[999] grid place-items-center w-full h-full bg-gray-950/75 dark:bg-gray-50/75 backdrop-blur-sm ease-in duration-300">
+          <div className="max-w-xl w-full mx-auto p-4 bg-white rounded-lg shadow-lg">
+            <div className="flex items-center justify-between">
+              <h2 className="font-semibold leading-tight text-gray-900"> {tr("report")->str} </h2>
+              <button
+                onClick={updateShowReport(setShowReport, false)}
+                className="w-6 h-6 flex items-center justify-center rounded-md text-gray-700 bg-gray-100 hover:bg-gray-200 hover:text-gray-900 focus:outline-none focus:ring-2 focus:ring-focusColor-500 focus:ring-offset-2 transition">
+                <Icon className="if i-times-light text-xl if-fw" />
+                <span className="sr-only"> {tr("close")->str} </span>
+              </button>
+            </div>
+            <textarea
```

Comment:
@pupilfirst/designers In dark mode, typed text is not visible - it appears to be black on black.

Reply:
@harigopal 
Resolved text color issue in dark mode. https://github.com/pupilfirst/pupilfirst/pull/1531/commits/ef6591cd40f5f4487360830825711a7139131642